common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -188,6 +212,31 @@ public void onFailure(Throwable t) {
     }
   }
 
+  private <T> void retry(
```

Comment:
Can we create a test suite for `BlockStoreClient` with UTs for this. Earlier this class didn't have much implementation. Since we have added this logic here, I think we can add UTs for it.